docs/simpledroidble/usage.rst

```diff
@@ -9,39 +9,103 @@ This code is currently under active development and some features are not yet
-        substitute module("org.simpleble.android:simpledroidble") with project(":simpledroidble")
+    includeBuild("path/to/simpledroidble") {
+        dependencySubstitution {
+            substitute module("org.simpleble.android:simpledroidble") with project(":simpledroidble")
+        }
     }
-}
 
-```kotlin
-includeBuild("path/to/simpledroidble") {
-    dependencySubstitution {
-        substitute(module("org.simpleble.android:simpledroidble")).using(project(":simpledroidble"))
+Or, in Kotlin::
+
+    includeBuild("path/to/simpledroidble") {
+        dependencySubstitution {
+            substitute(module("org.simpleble.android:simpledroidble")).using(project(":simpledroidble"))
+        }
     }
-}
-```
-
-Then, inside your `build.gradle` or `build.gradle.kts` file, you can add the
-following dependency:
-
-```groovy
-dependencies {
-    implementation "org.simpleble.android:simpledroidble"
-}
-```
-
-```kotlin
-dependencies {
-    implementation("org.simpleble.android:simpledroidble")
-}
-```
\ No newline at end of file
+
+Then, inside your ``build.gradle`` or ``build.gradle.kts`` file, you can add the
+following dependency::
+
+    dependencies {
+        implementation "org.simpleble.android:simpledroidble"
+    }
+
+Kotlin::
+
+    dependencies {
+        implementation("org.simpleble.android:simpledroidble")
+    }
+
+Using SimpleDroidBLE
+====================
+
+First you'll need to provide ``SimpleDroidBLE.contextReference`` with a WeakReference to the app's context. In this step, the native code will be loaded in memory and permissions will be requested to the user.
+
+The Adapter class
+-----------------
+
+The main entry point to SimpleDroidBLE is the Adapter object. You can get an Adapter instance via the static ``Adapter#getAdapters`` method, where the device's main bluetooth adapter will be returned.
+
+When using SimpleDroidBLE make sure to only reference a single instance of the Adapter object, as every instance creates a new internal referece in the native code.
+Having multiple instances (by unbounded calls to ``Adapter#getAdapters``, for example) may cause a memory leak.
+This can be accomplished by acquiring said instance either on your app's Application class, or, in a single activity app, on your Activity's viewmodel; and treating the Adapter as a global, singleton object.
+For an specific example you can check out `the sample app <https://github.com/OpenBluetoothToolbox/SimpleBLE/tree/main/examples/simpleble-android>`_.
+
+Connecting to devices
+---------------------
+
+SimpleDroidBLE is a `flow <https://kotlinlang.org/api/kotlinx.coroutines/kotlinx-coroutines-core/kotlinx.coroutines.flow/-flow/>`_-driven interface.
+
+That means that when looking to connect to a device, you should ``collect`` from ``Adapter#onScanFound`` and then start scanning for devices with ``Adapter#scanStart``. You'll get all found ``Peripheral`` s on said flow, where you can ``Peripheral#connect`` to whichever you need.
+
```

Comment:
This is good as a very basic approach, but the text needs to go deeper into detail, including examples that should be easy to follow for non-experts.